learning-progression.md

```diff
@@ -299,12 +299,17 @@ basic.forever(function () {
 ```
 
 #### 1. Study
-
+In the event a [button]{https://makecode.microbit.org/reference/input} a pressed all code is stopped. 
```

Comment:
This is throwing the students off the deep end. Waaaay to many assumptions.

_The processor often has to respond to events that happen outside of it (aka external events). The processor does not usually know what these are and when to expect them. The mechanism for communicating external events to the processor is called an **interrupt**. An interrupt is most often an electric signal sent over a wire._
_The pressing of a button of the micro:bit is one external event that needs to be communicated to the processor, so it generates an interrupt which is received by the processor._
_It is up to the programmer to tell the processor what to do when it detects an interrupt from the button. This is done by registering an **interrupt handler** with the interrupt. A handler is just a function._
_The programmatic way to register a handler for the button press is `input.onButtonPressed(...)`._
_The event "button pressed" is composite, in the sense that it contains two different simpler events that have been detected one after the other, namely "button depressed" and "button released"._
...